Adding complex example for event-based trigger using an icon template (#22021)
home-assistant/home-assistant.github.io: source/_integrations/template.markdown

{% endraw %}

> A more advanced use case could be to set the icon based on the sensor's own state like above, but when triggered by an event. This example demonstrates a binary sensor that turns on momentarily, such as when a doorbell button is pressed. When the button gets pressed, the first trigger will be triggered and the binary sensor's state will be updated. This in turn will trigger the second trigger, which will cause the icon template to be evaluated. The platform check in the state template is to avoid the update from going into an infinite loop of updating. 

Reviewer: This needs to be adjusted to the new situation :)

Author: Bussy week... :-) 

I updated it now. 

I dropped the infinite loop part, because it doesn't seem relevant at all now.

{% raw %}

```yaml

> template:
>   - trigger:
>       - platform: event
>         event_type: YOUR_EVENT
>       - platform: state
>         entity_id: binary_sensor.doorbell_rang
>         to: ~
>     binary_sensor:
>       name: doorbell_rang
>       icon: "{{ 'mdi:bell-ring-outline' if is_state('binary_sensor.doorbell_rang', 'on') else 'mdi:bell-outline' }}"
>       state: "{{ trigger.platform == 'event' or states('binary_sensor.doorbell_rang') }}"
>       auto_off:
>         seconds: 5

Reviewer: Wouldn't this be better?

```suggestion
template:
  - trigger:
      - platform: event
        event_type: YOUR_EVENT
      - platform: state
        entity_id: binary_sensor.doorbell_rang
        to: "off"
    binary_sensor:
      name: doorbell_rang
      icon: "{{ (trigger.platform == 'event') | iif('mdi:bell-ring-outline', 'mdi:bell-outline') }}"
      state: "{{ trigger.platform == 'event' }}"
      auto_off:
        seconds: 5
```

It triggers the exact changes that are relevant. The event and the auto turn off, instead of triggering all state changes. It also removes the need to query its own state (at that can already be determined as the state template does. Lastly, it only triggers 1 update for the template entity when the doorbell rings (icon & state, in that case, are rendered in a single update).

Author: Thanks for the suggestion! It reads a lot easier to me – and I can see it being more efficient.
